### PR TITLE
ref(metrics layer) Fix HAVING clause to resolve metrics

### PR DESCRIPTION
**Pull request by evanh**
The having clause was not previously being resolved, which meant queries would
fail because the MRI's were not being resolved.

Add a function to the builder that resolves the expressions in the HAVING
clause. The HAVING clause is effectively made up of conditions on SELECT
expressions, so use the same functionality as the SELECT to resolve the LHS of
the HAVING conditions.

The functionality is very limited right now (no AND/OR etc.), but that is
partly by design. Supporting the HAVING clause is a workaround and it shouldn't
be too permissive.

### CODE REVIEW

File: src/sentry/snuba/metrics/query_builder.py [hunk 1]
```diff
@@ -925,6 +925,37 @@ def _build_orderby(self) -> Optional[List[OrderBy]]:
 
         return orderby_fields
 
+    def _build_having(self) -> List[Union[BooleanCondition, Condition]]:
```
Comment (iambriccardo) [inline src/sentry/snuba/metrics/query_builder.py:928]: Nice catch!

The thing is that, do we want to support all derived metrics in the having?

Comment (evanh) [reply src/sentry/snuba/metrics/query_builder.py:928]: I don't really want to support HAVING at all, but I don't see a reason to restrict the derived metrics.

Comment (iambriccardo) [reply src/sentry/snuba/metrics/query_builder.py:928]: We actually backtracked on the decision to support having and wanted to delete it. There was a miscommunication between teams that requested us to implement it. If I am correct, it's not used.

Comment (evanh) [reply src/sentry/snuba/metrics/query_builder.py:928]: It is used, certainly in a number of tests. Hence why I have this PR. You can see which tests here: https://github.com/getsentry/sentry/pull/55080/files#diff-126943646d55760ffab2a29b719f2a6b0cf2d14c5a65ff1c65fb46e7561f5e42L2460

**Review by enochtangg — APPROVED** [review]

**Pull request merged**